src/main/java/com/github/lzyzsd/circleprogress/DonutProgress.java

```diff
@@ -116,19 +159,40 @@ private int measure(int measureSpec){
 		return result;
 	}
 
+	public static float round(float d, int decimalPlace) {
+		BigDecimal bd = new BigDecimal(Float.toString(d));
```

Comment:
Creating a new `BigDecimal` every time we receive some data is quite inefficient -- the conversion from bytes to MB could be done purely using integer arithmetic, e.g.:

		final int totalKilobytes = bytes / 1024;
	    final int totalMegabytes = totalKilobytes / 1024;
	    final int remainder = totalKilobytes % 1024;
	    final String result = String.format("%d.%02d MB", totalMegabytes, (remainder / 10));

The above should produce a result in the form "2.35 MB".